### PR TITLE
refactor: to update to jest v27

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

~Quick attempt to update to Jest v27 to triage impact.~

Updates to Jest v27 including sibling libraries while remaining backwards compatible.

### CODE REVIEW

**Comment by tdeekens** [comment]
We will have to wait a bit for the community to work its way towards it just like we did here https://github.com/kentcdodds/kcd-scripts/pull/210.

File: packages/jest-preset-mc-app/polyfills/intl.js [hunk 1]
```diff
@@ -4,3 +4,5 @@
 // https://github.com/formatjs/formatjs/tree/master/packages/intl-messageformat#modern-intl-dependency
 global.Intl = require('intl');
 global.Intl.PluralRules = require('./intl-pluralrules');
+// Work-around for the issue  https://github.com/andyearnshaw/Intl.js/issues/256
+global.Intl.__disableRegExpRestore();
```
Comment (tdeekens) [inline packages/jest-preset-mc-app/polyfills/intl.js:8]: Good one!

File: packages/jest-preset-mc-app/transform-babel-jest.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 const getBabePresetConfigForMcApp = require('@commercetools-frontend/babel-preset-mc-app');
 const getJestBabelPreset = require('babel-preset-jest');
-const babelJest = require('babel-jest');
+const { default: babelJest } = require('babel-jest');
```
Comment (tdeekens) [inline packages/jest-preset-mc-app/transform-babel-jest.js:3]: This is likely breaking. If consumers don't update their Jest this won't work. Maybe we have to sniff out the babel-jest version from the `package.json`. Not super sure yet.

File: packages/jest-preset-mc-app/transform-babel-jest.js [hunk 3]
```diff
@@ -14,6 +14,22 @@ const jestBabelConfig = {
...
+ *  In `babel-jest` v27 the export has changed
+ *  from: `babelJest.createTransformer`
+ *  to `babelJest.default.createTransformer`.
+ *  In order to remain backwards compatible the
+ *  type of the export is checked.
+ *
+ *  TODO: v21: Remove check and release `jest-preset-mc-app`
+ *  as a breaking release.
+ */
+const createTransformer =
```
Comment (pa3) [inline packages/jest-preset-mc-app/transform-babel-jest.js:28]: Nice idea!

**Review by pa3 — APPROVED** [review]
Looks good to me. 👍

**Pull request merged**